ElrondNetwork/elrond-go
# Protocol tx fields

## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- When querying for `ApiTransactionResult` (either in block/proxy hyperblock), some protocol transaction+scr fields were missing. Some of them were mandatory for external clients to re-create protocol transaction(e.g. validate transaction signature based on transaction version + options). 
## Proposed Changes
- Added  missing fields support for API calls
## Testing procedure
`ApiTransactionResult` holds all type of txs in a single struct (move balance, scr, rewards, invalid, etc).
To test it, generate different types of transactions and check the validity of the following:
- `ChainID + Options` : should be available and **correct** for all tx types 
- `Version`: should be available only for normal txs, otherwise it should be missing
- `CallType` + `RelayerAddress` + `RelayedValue`: for scr txs, should be correctly computed, otherwise these should be missing

An observer node from each shard should be individually queried for these information, as well as proxy(hyperblock for example). For proxy, one should use this branch: `EN-13246-protocol-tx-fields`
